app/controllers/timelines_controller.rb

```diff
@@ -360,6 +353,20 @@ def load_timeline_token
       current_ability.options[:timeline_token] = @timeline_token
     end
 
+    def build_url_fragment
+      url_fragment = "noHeader=true&noFooter=true&noSourceLink=false&noVideo=false"
+
+      if @timeline.visibility == 'public' || current_user == @timeline.user
+        url_fragment += "&resource=#{Addressable::URI.escape_component(manifest_timeline_url(@timeline, format: :json), /[:\/?=]/)}"
+        url_fragment += "&callback=#{Addressable::URI.escape_component(manifest_timeline_url(@timeline, format: :json), /[:\/?=]/)}"
+      elsif current_user || @timeline.valid_token?(@timeline.access_token)
+        url_fragment += "&resource=#{Addressable::URI.escape_component(manifest_timeline_url(@timeline, format: :json, token: @timeline.access_token), /[:\/?=]/)}"
+        url_fragment += "&callback=#{Addressable::URI.escape_component(timelines_url, /[:\/?=]/)}"
+      end
```

Comment:
I think this might be clearer to pull apart the two query params:
```suggestion
     # Manifest url - include token if present 
      if @timeline_token.present?
        url_fragment += "&resource=#{Addressable::URI.escape_component(manifest_timeline_url(@timeline, format: :json, token: @timeline_token), /[:\/?=]/)}"
      else 
        url_fragment += "&resource=#{Addressable::URI.escape_component(manifest_timeline_url(@timeline, format: :json), /[:\/?=]/)}"
      end

      # Save callback - Owner uses manifest url and all other logged-in users use timeline index url
      if current_user == @timeline.user
        url_fragment += "&callback=#{Addressable::URI.escape_component(manifest_timeline_url(@timeline, format: :json), /[:\/?=]/)}"
      elsif current_user
        url_fragment += "&callback=#{Addressable::URI.escape_component(timelines_url, /[:\/?=]/)}"
      end
```
Note I also switched `@timeline.access_token` to `@timeline_token` to be clearer that it is the param.
Also the save callback requires a logged in user.